src/components/Messages/index.tsx

```diff
@@ -1,13 +1,95 @@
+import { Card } from '@components/UI/Card';
 import { GridItemEight, GridItemFour, GridLayout } from '@components/UI/GridLayout';
 import MetaTags from '@components/utils/MetaTags';
+import { Profile } from '@generated/types';
 import isFeatureEnabled from '@lib/isFeatureEnabled';
-import { NextPage } from 'next';
+import { Client, Conversation, Stream } from '@xmtp/xmtp-js';
+import { useRouter } from 'next/router';
+import { FC, useEffect, useState } from 'react';
 import { APP_NAME } from 'src/constants';
 import Custom404 from 'src/pages/404';
 import { useAppStore } from 'src/store/app';
+import { useXmtpStore } from 'src/store/xmtp';
+import { useAccount, useSigner } from 'wagmi';
 
-const Messages: NextPage = () => {
+interface Props {
+  profile: Profile;
+}
+
+const Messages: FC<Props> = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
   const currentProfile = useAppStore((state) => state.currentProfile);
+  const [stream, setStream] = useState<Stream<Conversation>>();
+  const xmtpState = useXmtpStore((state) => state);
+  const { client, setClient, conversations, setConversations, messages, setMessages, setLoading } = xmtpState;
+  const router = useRouter();
+
+  useEffect(() => {
+    const initXmtpClient = async () => {
+      if (signer && !client) {
+        const xmtp = await Client.create(signer);
+        setClient(xmtp);
+      }
+    };
+    if (isFeatureEnabled('messages', currentProfile?.id)) {
+      initXmtpClient();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [signer]);
+
+  useEffect(() => {
+    if (!client || !isFeatureEnabled('messages', currentProfile?.id)) {
+      return;
+    }
+
+    async function listConversations() {
+      setLoading(true);
+      const convos = (await client?.conversations?.list()) || [];
+      Promise.all(
+        convos.map(async (convo) => {
+          if (convo.peerAddress !== address) {
+            const newMessages = await convo.messages();
+            messages.set(convo.peerAddress, newMessages);
+            setMessages(new Map(messages));
+            conversations.set(convo.peerAddress, convo);
+            setConversations(new Map(conversations));
+          }
+        })
+      ).then(() => {
+        setLoading(false);
+      });
+    }
+    const streamConversations = async () => {
+      const newStream = (await client?.conversations?.stream()) || [];
+      setStream(newStream);
+      for await (const convo of newStream) {
+        if (convo.peerAddress !== address) {
+          const newMessages = await convo.messages();
+          messages.set(convo.peerAddress, newMessages);
+          setMessages(new Map(messages));
+          conversations.set(convo.peerAddress, convo);
+          setConversations(new Map(conversations));
+        }
+      }
+    };
+    listConversations();
+    streamConversations();
+    return () => {
```

Comment:
```suggestion
    listConversations();
    streamConversations();

    return () => {
```

suggested to leave new line before return 🚀